demo 2/src/main/java/com/example/demo/service/AccountService.java

```diff
@@ -0,0 +1,155 @@
+        }
+
+        int number = String.valueOf(account.getPhone()).length();
+        if (number > 10 || number < 10) {
+            log.warn("Enter 10 digit phone number");
+            throw new DuplicateKeyException(ResultInfoConstants.ENTER_10_DIGIT_PHONE_NUMBER);
+        }
+        int password = String.valueOf(account.getPassword()).length();
+        if (password > 4 || password < 4) {
+            log.warn("Enter 4 digit password");
+            throw new DuplicateKeyException(ResultInfoConstants.ENTER_4_DIGIT_Password);
+        }
+        Long num = account.getPhone();
+        AccountTable accountTable = accountRepository.getUser(num);
+        if (accountTable != null) {
+            log.warn("Invalid phone number");
+            throw new AccountNotFoundException(ResultInfoConstants.ACCOUNT_ALREADY_EXISTING);
+        }
+        OtpTable otpTable = otpRepository.getUser(num);
+        if (otpTable == null) {
+            log.warn("Generate otp");
+            throw new DuplicateKeyException(ResultInfoConstants.GENERATE_OTP);
+        }
+        if (account.getOtp() != otpTable.getOtp()) {
+            log.warn("Put valid otp");
+            throw new DuplicateKeyException(ResultInfoConstants.PUT_VALID_OTP);
+        }
+        otpRepository.deleteById(num);
+        return accountRepository.save(account.toAccountTable()).getPhone();
+    }
+
+    public Long resetPassword(Account account) {
+
+        if (account.getPhone() == null || account.getPassword() == 0) {
+            log.warn("Enter phone number and password");
+            throw new DuplicateKeyException(ResultInfoConstants.ENTER_PHONE_NUMBER_AND_PASSWORD);
+        }
+
+        int number = String.valueOf(account.getPhone()).length();
+        if (number > 10 || number < 10) {
+            log.warn("Enter 10 digit phone number");
+            throw new DuplicateKeyException(ResultInfoConstants.ENTER_10_DIGIT_PHONE_NUMBER);
+        }
+        int password = String.valueOf(account.getPassword()).length();
+        if (password > 4 || password < 4) {
+            log.warn("Enter 4 digit password");
+            throw new DuplicateKeyException(ResultInfoConstants.ENTER_4_DIGIT_Password);
+        }
+
+        Long num = account.getPhone();
+        AccountTable accountTable = accountRepository.getUser(num);
+        if (accountTable == null) {
+            log.warn("Invalid phone number");
+            throw new AccountNotFoundException(ResultInfoConstants.ACCOUNT_NOT_FOUND);
+        }
+
+        OtpTable otpTable = otpRepository.getUser(num);
+        if (otpTable == null) {
+            log.warn("Generate otp");
+            throw new DuplicateKeyException(ResultInfoConstants.GENERATE_OTP);
+        }
+
+        if (account.getOtp() != otpTable.getOtp()) {
+            log.warn("Put valid otp");
+            throw new DuplicateKeyException(ResultInfoConstants.PUT_VALID_OTP);
+        }
```

Comment:
otp info should not be in account.

Reply:
we are not validating otp received from request here

Reply:
Instead of in memory check
```
Otp otp = getup(..)
if(otp != x) {
  ...
}
```

Need to write a query and in where clause pass the received otp. If it returns empty or null then the otp is not matched